Sources/NIO/Codec.swift

```diff
@@ -96,59 +96,83 @@ private extension ChannelHandlerContext {
 
 extension ByteToMessageDecoder {
 
+    /// Decode in a loop until there is nothing more to decode.
+    private func decodeLoop(ctx: ChannelHandlerContext, decodeFunc: (ChannelHandlerContext, inout ByteBuffer) throws -> DecodingState) {
+        assert(self.cumulationBuffer != nil)
+        ctx.withThrowingToFireErrorAndClose {
```

Comment:
Probably better to hoist this up to the caller.

Reply:
You mean let the error bubble up ?

Reply:
Let `decodeLoop` throw, place the `ctx.thenThrowingToFireErrorAndClose` in the function that calls `decodeLoop`.